roles/velero/templates/velero_bsl_gcp.yml.j2

```diff
@@ -0,0 +1,33 @@
+---
+{% for bsl in backup_storage_locations %}
+{% if bsl.provider == 'gcp' %}
+apiVersion: velero.io/v1
+kind: BackupStorageLocation
+metadata:
+  name: {{ bsl.name }}
+  namespace: {{ velero_namespace }}
+spec:
+  provider: {{ bsl.provider }}
+  objectStorage:
+    bucket: {{ bsl.object_storage.bucket }}
+{% if bsl.object_storage.prefix is defined %}
```

Comment:
Just curious, do we need all of these `if` checks? What happens if we just specify them and the fields don't exist? Does it break or do we just set empty values?

Reply:
The creation of the resource using the template breaks. For instance, if you are configuring BSL via AWS and lets assume that the end user did not give any `config` related props in the CR as well as there are no `if defined` checks for around `config` in the template, the resource creation will fail, it will state that the `dict object` has no attribute `config` and this would result in failure of the ansible task.

Reply:
Awesome, thanks for the info.